pkg/generate/code/compare.go

```diff
@@ -562,7 +562,7 @@ func CompareStruct(
 		secondResAdaptedVarName := secondResVarName + "." + memberNameClean
 
 		var compareConfig *ackgenconfig.CompareFieldConfig
-		fieldConfig := fieldConfigs[memberFieldPath]
+		fieldConfig := fieldConfigs[memberFieldPath[5:len(memberFieldPath)]]
```

Comment:
What's this for? (Note: if I'm asking this question, pretty sure other readers of the code will ask the same question without a code comment explaining why... :) )

Reply:
Agreed, updated it

memberFieldPath contains the field path along with the prefix `Spec.` hence we would need to substring to exclude `Spec.` to get correct field config.

https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/generate/code/compare.go#L128 adds `Spec.` prefix.